src/stdatm/atmosphere_partials.py

```diff
@@ -0,0 +1,194 @@
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from typing import Sequence, Union
+
+import numpy as np
+from scipy.constants import R, atmosphere, foot
+
+from .atmosphere import Atmosphere
+
+from .partials_state_parameters import (
+    compute_partial_temperature,
+    compute_partial_pressure,
+    compute_partial_density,
+    compute_partial_speed_of_sound,
+    compute_partial_dynamic_viscosity,
+    compute_partial_kinematic_viscosity,
+)
+
+AIR_MOLAR_MASS = 28.9647e-3
+AIR_GAS_CONSTANT = R / AIR_MOLAR_MASS
+SEA_LEVEL_PRESSURE = atmosphere
+SEA_LEVEL_TEMPERATURE = 288.15
+TROPOPAUSE = 11000
+
+
+class AtmospherePartials(Atmosphere):
+    """
+    Implementation of International Standard Atmosphere for troposphere and stratosphere with
+    derivatives of state parameters with respect to altitude.
+
+    Atmosphere properties and partials are provided in the same "shape" as provided
+    altitude:
+
+    - if altitude is given as a float, returned values will be floats
+    - if altitude is given as a sequence (list, 1D numpy array, ...), returned
+      values will be 1D numpy arrays
+    - if altitude is given as nD numpy array, returned values will be nD numpy
+      arrays
+
+    The AtmospherePartials class inherits from the Atmosphere class and thus retains its usages. It
+    however adds the computation of the partial derivatives of all state properties with respect to
+    the altitude.
+
+    Usage:
+
+    .. code-block::
+        >>> from stdatm import AtmospherePartials
+        >>> pressure = AtmospherePartials(30000).pressure # pressure at 30,000 feet, dISA = 0 K
+        >>> partials_pressure_altitude = AtmospherePartials(30000).partial_pressure_altitude # pressure at 30,000 feet, dISA = 0 K
+
+        >>> atm = AtmospherePartials([0.0,10000.0,30000.0]) # init for alt. 0, 10,000 and 30,000 feet
+        >>> atm.partial_pressure_altitude # derivative of pressures with respect to altitude for all defined altitudes
+        array([-3.66160356, -2.70401861, -1.36992549])
+        >>> atm.partial_dynamic_viscosity_altitude # derivative of dynamic viscosities with respect to altitude for all defined altitudes
+        array([-9.55961630e-11, -9.88873356e-11, -1.06349854e-10])
+    """
+
+    def __init__(
+        self,
+        altitude: Union[float, Sequence[float]],
+        delta_t: float = 0.0,
+        altitude_in_feet: bool = True,
+    ):
+        """
+        :param altitude: altitude (units decided by altitude_in_feet)
+        :param delta_t: temperature increment (°C) applied to whole temperature profile
+        :param altitude_in_feet: if True, altitude should be provided in feet. Otherwise,
+                                 it should be provided in meters.
+        """
+
+        # Can't use **kwargs because we need to define unit_coeff as a an attribute for the
+        # computation of partials
```

Comment:
Removed the comment in cedae02.
As for `AtmosphereSIWithPartials`, it would be easy now that we have the more generic `AtmosphereWithPartials`. Since it works for both case, I think we can leave it like that but if you think it's worth creating a new class, if only to match what is done `Atmosphere`, I can add it. You tell me :smile: